ServerCore/ModelBases/EventSpecificPageModel.cs

```diff
@@ -129,7 +129,8 @@ public async Task<bool> IsEventAdmin()
         /// </summary>
         public async Task<bool> HasSwag()
         {
-            return Event.HasSwag && await _context.PlayerInEvent.Where(m => m.Event == Event && m.Player == LoggedInUser).AnyAsync();
+            PlayerInEvent playerSwag = await _context.PlayerInEvent.Where(m => m.Event == Event && m.Player == LoggedInUser).FirstOrDefaultAsync();
+            return Event.HasSwag && (playerSwag != null) && !string.IsNullOrWhiteSpace(playerSwag.Lunch);
```

Comment:
We can early exit w/o hitting the DB if (!Event.HasSwag) - which is arguably the most common case.

Reply:
I just looked, and apparently this is only called once inside an if block that's triggered when the event does have swag, so that Event.HasSwag ends up being unnecessary since it will ALWAYS be true.